### PR TITLE
[MXNET-1253] [WIP] Fix large array issue in topk operator

### PR DESCRIPTION
**Pull request by apeforest**
## Description ##
This is a second batch of fixes for several sorting operators.

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [X] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [X] Changes are complete (i.e. I finished coding on this PR)
- [X] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [X] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [X] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [X] topk
- [X] test_large_array.py

## Comments ##
- Removed dtype for topk operator for indices. It seems unnecessary (and confusing) to have float32 as a type for indices and not clear why user would ever want to define it. This may cause backward compatibility issue only if users set dtype for return indices of a topK operator.

### CODE REVIEW

**Comment by Roshrini** [comment]
@apeforest Thanks for working on this.
@mxnet-label-bot Add [pr-work-in-progress, Operator]

**Comment by Roshrini** [comment]
@apeforest Can you address comments and rebase this PR? Thanks

File: src/operator/tensor/ordering_op-inl.h [hunk 1]
```diff
@@ -58,7 +58,6 @@ struct TopKParam : public dmlc::Parameter<TopKParam> {
   int k;
   int ret_typ;
   bool is_ascend;
-  int dtype;
```
[the remark this answers is not in the document]
Comment (zheng-da) [reply src/operator/tensor/ordering_op-inl.h]: agree. could you please revert the change?

**Comment by stu1130** [comment]
@apeforest ping again for the update

**Comment by apeforest** [comment]
Got fully occupied with another project. I will resume this PR when I get some slack next week.

**Comment by ankkhedia** [comment]
@apeforest Did you get a chance to work on this PR?
#13627 depends on this fix.

**Comment by pinaraws** [comment]
@apeforest  Any updates?

**Comment by piyushghai** [comment]
@apeforest Can you rebase with the current master branch and look into the CI failures ?

**Comment by Roshrini** [comment]
@apeforest Any update on this PR?

**Comment by pinaraws** [comment]
@apeforest Any update on this PR?

**Comment by apeforest** [comment]
@access2rohit is working on this in a separate new PR. Closing this one.

**Pull request closed without merging**